src/ILCompiler.Compiler/src/Compiler/VirtualMethodCallHelper.cs

```diff
@@ -83,7 +83,11 @@ public static int GetGenericDictionarySlot(NodeFactory factory, TypeDesc type)
         /// </summary>
         public static bool HasGenericDictionarySlot(this TypeDesc type)
         {
-            return !type.IsInterface && type.HasInstantiation &&
+            // Dictionary slots on generic interfaces are necessary to support static methods on interfaces
+            if (type.IsInterface)
```

Comment:
Just remove the `type.IsInterface` from the return below. Otherwise you're forcing dictionary slots for interfaces even if we're not compiling shared code, or when interface methods wouldn't need a dictionary anyway (e.g. `IEnumerable<int>`).